File: 1765-merge-in-between-linked-lists/1765-merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode curr1 = list1;
        ListNode curr2 = list1;
        ListNode curr3 = list2;
        //Iterate list1 until you reach the node at a-1 so that you can point the next of the curr1 to the head of the list2
        for(int i=0;i<a-1;i++){
            curr1 = curr1.next;
        }
        //Iterate list1 until you reach the node at b+1 so that you can point the last node of the list2 to the curr2
        for(int i=0;i<=b;i++){
            curr2 = curr2.next;
        }

        //find the last node of the list2
        while(curr3.next != null){
            curr3 = curr3.next;
        }

        //(from the testcase1 point curr1(13).next to head of the list2(1000000))
        curr1.next = list2;
        //point the last node of the list2's next(1000002) to the curr2(5) 
        curr3.next = curr2;
        //return the updated list1
        return list1;



    }
}

File: 1765-merge-in-between-linked-lists/1765-merge-in-between-linked-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeInBetween(struct ListNode* list1, int a, int b, struct ListNode* list2){
    struct ListNode *current1=list1;//10....
    struct ListNode *current2=list1;//10....
    struct ListNode *current3=list2;//100000...
    //loop for getting the head of the list2 attaching to a-1
    for(int i=0;i<a-1;i++){//0 1
        current1=current1->next;//iterate until it reaches before a-1
        // 1 13
    }
    //from example current1-->13
    //loop for getting the tail of the list2 after b nodes and to attach the tail of list 2 to this node
    for(int i=0;i<=b;i++){//0 1 2 3 4
        current2=current2->next;//1 13 6 9 5
        //current2-->5
    }

    //until list2 reaches the last element
    while(current3->next!=NULL){
        current3=current3->next;//1000002
    }
    //
    current1->next=list2;//13->next=1000000
    current3->next=current2;//1000002->next=5
    return list1;
}